### DISCUSSION TITLE
all text way of giving settings to nodes or even adding nodes to the node tree

### DISCUSSION THREAD

**Discussion by thecodsman** [godotengine/godot-proposals#6542] [Editor]
so basically you could copy and paste a node tree or node settings from only text (kind of like config files). I think the benefits are pretty self-explanatory if you needed help you dont need to manually list out the settings or send a screenshot (if its on something that only supports text or you dont have image perms) you can copy and paste node settings from a tutorial so you know its the right node settings yeah.

**Comment by Calinou** [comment]
This already exists in the form of `.tscn` and `.tres` files :slightly_smiling_face:

**Reply by thecodsman** [reply]
oh I didnt realize those were human-readable. thanks!
